.github/workflows/benchmark_execution.yml

```
@@ -67,12 +67,17 @@ jobs:
         run: ./build_tools/scripts/git/update_runtime_submodules.sh
       - name: "Generating benchmark matrix"
         id: generate
-        run: |
-          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}"
-          echo benchmark-matrix=$(jq \
-              'to_entries | map({"device_name": .key, "host_environment": .value.host_environment})' \
-              "${BENCHMARK_CONFIG}") \
-            >> "${GITHUB_OUTPUT}"
+        run: >
+          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}";
+          echo benchmark-matrix=$(jq '[ . | to_entries[]
+            | .key as $device_name
+            | .value.host_environment as $host_environment
+            | (.value.shards | length) as $shard_count
+            | .value.shards[]
+            | { device_name: $device_name, host_environment: $host_environment,
```

There's a shorthand syntax when you just want to expand a variable to its key and value. You could consider using a consistent name for `shard_index` (i.e. calling it `index` in the original config) to even further take advantage of that.

```suggestion
            | {$device_name, $host_environment,
```

---

Cool. Didn't know about that feature. Thx.